### PR TITLE
py-scikit-learn: update to 1.5.2

### PR DESCRIPTION
**Pull request by i0ntempest**
temporary workaround for numpy v2 requirement

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by i0ntempest** [comment]
Only 1.5+ can be built on macOS 15.

**Comment by i0ntempest** [comment]
The only one failing is python 3.8 subport, which cannot be updated. Should I just remove it? Like in 10 days python 3.8 will be EOL anyway.

**Comment by reneeotten** [comment]
> The only one failing is python 3.8 subport, which cannot be updated. Should I just remove it? Like in 10 days python 3.8 will be EOL anyway.

You can (and I would be in favor), just make sure there are no dependents and if they are you'll need to go down the dependency tree to remove PY38 subport from all relevant ports.

File: python/py-scikit-learn/Portfile [hunk 1]
```diff
@@ -24,43 +25,37 @@ long_description    Scikit-learn integrates machine learning algorithms \
...
-                    size    7743688
+checksums           rmd160  c44852ef8d63ca0da228bc7a3850ef99900568a9 \
+                    sha256  b4237ed7b3fdd0a4882792e68ef2545d5baa50aca3bb45aa7df468138ad8f94d \
+                    size    7001680
 
 if {${name} ne ${subport}} {
     depends_build-append \
-                        path:bin/cython-${python.branch}:py${python.version}-cython
+                        path:bin/cython-${python.branch}:py${python.version}-cython \
+                        port:py${python.version}-meson-python
```
Comment (reneeotten) [inline python/py-scikit-learn/Portfile:35]: instead please just set `python.pep517_backend meson`

**Comment by i0ntempest** [comment]
> > The only one failing is python 3.8 subport, which cannot be updated. Should I just remove it? Like in 10 days python 3.8 will be EOL anyway.
> 
> You can (and I would be in favor), just make sure there are no dependents and if they are you'll need to go down the dependency tree to remove PY38 subport from all relevant ports.

Opened a separate PR for that: https://github.com/macports/macports-ports/pull/25858

**Pull request merged**